src/partition/repart.c

```diff
@@ -3022,14 +3023,6 @@ static int partition_encrypt(
 
         log_info("Encrypting future partition %" PRIu64 "...", p->partno);
 
-        volume_key = malloc(volume_key_size);
-        if (!volume_key)
-                return log_oom();
-
-        r = crypto_random_bytes(volume_key, volume_key_size);
-        if (r < 0)
-                return log_error_errno(r, "Failed to generate volume key: %m");
```

Comment:
why do this btw?

for use in confidential computing it might be useful to pass in the volume key (read from a systemd credential for example), or derive it from the configured seed.

Reply:
(the thing is that we'll now measure it via systemd-cryptsetup, and hence there's quite some value in being able to pre-provision it)

Reply:
libcryptsetup maintainer recommended we let libcryptsetup take care of generating the volume key instead of doing it ourselves. If I understand correctly, aside from changing the volume key size, this commit should not change any other behavior.